Tasks/ServiceFabricDeploy/ServiceFabricSDK/Utilities.ps1

```diff
@@ -204,4 +204,23 @@ function Merge-HashTables
     }
     $HashTableNew = $HashTableOld + $HashTableNew
     return $HashTableNew
+}
+
+function Get-TempDirectoryPath {
+    <#
+    .SYNOPSIS
+    Returns a temp directory path. Uses Agent.TempDirectory if available
+    #>
+
+    Param ()
+
+    $agentVersion = Get-VstsTaskVariable -Name 'agent.version'
+    if (!$agentVersion -or (([version]'2.115.0').CompareTo([version]$agentVersion) -ge 1))
```

Comment:
This check just make sure the agent version is greater than 2.115.0 and the variable Agent.TempDirectory is defined since then. However, don't we need to check if Agent.TempDirectory is shorter than env:Temp for real?

Reply:
My assumption is that agent directory will mostly be smaller than temp directory. However, better to not make any assumption. Updated code.